docs/implementation_design/object_storage.md

```diff
@@ -1 +1,87 @@
 # Object storage
+In GoShimmer `ObjectStorage`  is used as a base data structure for many data collections elements such as `branchStorage`, `conflictStorage`, `messageStorage` and others.
+It can be described by the following characteristics, it:
+-  is a manual cache which keeps objects as long as consumers are using it.
+- uses key-value storage type, defines its object factory, and mutex options for guarding shared variables and preventing changing object state by multiple gorutines at the same time.
```

Comment:
(from the sentence wasnt clear to me if object factory guards and prevents also the latter stuff? if not then write)
... defines its object factory, and defines mutex options for guarding shared variables and preventing ...

the object state
goroutines

Reply:
corrected